gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -269,20 +296,18 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
             kspTask.commandLineArgumentProviders.addAll(kspExtension.commandLineArgumentProviders)
 
             kspTask.options.addAll(
-                kspTask.project.provider {
-                    getSubpluginOptions(
-                        project,
-                        kspExtension,
-                        sourceSetName,
-                        target,
-                        isIncremental,
-                        kspExtension.allWarningsAsErrors,
-                        kspTask.commandLineArgumentProviders,
-                        emptyList(),
-                    )
-                }
+                getSubpluginOptions(
+                    project = project,
+                    kspExtension = kspExtension,
+                    sourceSetName = sourceSetName,
+                    target = target,
+                    isIncremental = isIncremental,
+                    allWarningsAsErrors = project.provider { kspExtension.allWarningsAsErrors },
```

Comment:
Ideally `allWarningsAsErrors` should be a a property already but for now this is better.